DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -27,11 +43,44 @@ public IActionResult Index()
             var userDelegateId = User.GetCustomClaim(CustomClaimTypes.LearnCandidateId);
             var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
 
-            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(delegateUser?.CentreId);
-
-            var model = new MyAccountViewModel(adminUser, delegateUser, customPrompts);
+            var model = new EditDetailsViewModel(adminUser, delegateUser);
 
             return View(model);
         }
+
+        [HttpPost]
+        public IActionResult EditDetails(EditDetailsViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var userAdminId = User.GetCustomClaim(CustomClaimTypes.UserAdminId);
+            var userDelegateId = User.GetCustomClaim(CustomClaimTypes.LearnCandidateId);

```

Comment:
I think you should be able to use User.GetCandidateId() here (as per CustomClaims extension). I think Jon B added another extension method for the admin equivalent, which should be on the latest master (User.GetAdminId).